lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -834,8 +834,16 @@ def install_puppet_agent_from_dmg_on(hosts, opts)
         def get_latest_puppet_agent_build_from_url(url)
           require 'oga'
           require 'net/http'
-          body = Net::HTTP.get(URI.parse("#{url}/index_by_lastModified_reverse.html"))
-          document = Oga.parse_html(body)
+
+          full_url = "#{url}/index_by_lastModified_reverse.html"
+          response = Net::HTTP.get_response(URI(full_url))
+
+          # Redirect following
+          while response.code == '301'
+            response = Net::HTTP.get_response(URI.parse(Net::HTTP.get_response(URI(full_url))['location']))
```

Comment:
It's possible the client could get into an infinite redirect loop if the server is misbehaving. Might want to keep track of a redirect limit and then abort if we reach it, like https://stackoverflow.com/questions/6934185/ruby-net-http-following-redirects

Reply:
Great point!